src/inmanta/execute/runtime.py

```diff
@@ -267,6 +267,51 @@ def get_dataflow_node(self) -> dataflow.AssignableNodeReference:
         return self._node
 
 
+class ResultVariableProxy(VariableABC[T]):
```

Comment:
cunning.

Reply:
I guess it would be a step closer to a dataflow graph based execution, but I think we would need to be really careful about semantics: can you set a value on a proxy? Does it set it on that RV only or does it set it on the one it is connected to as well? What if a proxy is frozen? ...